src/pages/signup/index.jsx

```diff
@@ -1,7 +1,174 @@
-import React from 'react';
+import React, { useState } from 'react';
+import bcrypt from 'bcryptjs';
+import isError from '../../utils/signUpFuncs';
 
 const SignUp = () => {
-  return <div>Hello Signup</div>;
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [phone, setPhone] = useState('');
+  const [password1, setPassword1] = useState('');
+  const [password2, setPassword2] = useState('');
+  const [errors, setErrors] = useState({});
+  const [showNoti, setShowNoti] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const data = {
+      email,
+      username,
+      phone,
+      password1,
+      password2,
+    };
+    const err = isError(data);
+    setErrors(err);
+
+    if (Object.keys(err).length === 0) {
+      const hash = bcrypt.hashSync(password1, 8);
+
+      const users = JSON.parse(localStorage.getItem('users'));
+      const newUser = {
+        id: users.length,
+        email,
+        username,
+        password: hash,
+        phone,
+      };
+
+      users.push(newUser);
+      localStorage.setItem('users', JSON.stringify(users));
+
+      setShowNoti(true);
+    } else setShowNoti(false);
+  };
+
+  return (
+    <div className="h-screen bg-gray-100 flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
+      <div className="max-w-md w-full space-y-8">
+        {showNoti ? (
+          <div
+            className="p-4 mb-4 text-sm text-green-700 bg-green-100 rounded-lg dark:bg-green-200 dark:text-green-800"
+            role="alert">
+            <span className="font-medium">Sign up sucessfully!</span>{' '}
+            Refresh localStorage to see new account.
```

Comment:
What is the show noti purpose here?

Reply:
It's just an pop up alert when successfully signup for debugging purposes

Reply:
Hmmm for user perspective, we can delete this logic maybe ?

Reply:
Sure, this was just for debugging, we will redirect the user to the homepage after successfuly sign up

Reply:
We should just redirect the user to login screen for security purpose, and to make sure the user not forgetting their account

Reply:
Okay